### ISSUE TITLE
evaluate_uns: is it expected that z error is high when compute_z loss is low or average probability of the target token across batch is high?

### ISSUE THREAD

**Issue by jasonrichdarmawan** [jianghoucheng/AnyEdit#5]
Focus of the code:
https://github.com/jianghoucheng/AnyEdit/blob/ac8e65240fec5ada76ce37360727b5dcbb1d5672/memit_ARE/memit_ARE_main.py#L109

Command:
```
python3 -m experiments.evaluate_uns     --alg_name=MEMIT_ARE     --model_name=meta-llama/Meta-Llama-3-8B-Instruct     --hparams_fname=Llama3-8B-Instruct.json     --ds_name=unke     --dataset_size_limit=1000     --num_edits=1
```

Changes to code? `float32` to `float16` due to VRAM constraints (1x 3090)
```
        model = AutoModelForCausalLM.from_pretrained(
            model_name,
            device_map="auto",
            dtype=torch.float16,
        )
```
https://github.com/jianghoucheng/AnyEdit/blob/ac8e65240fec5ada76ce37360727b5dcbb1d5672/experiments/evaluate_uns.py#L86

```
    "mom2_dtype": "float16",
```
https://github.com/jianghoucheng/AnyEdit/blob/ac8e65240fec5ada76ce37360727b5dcbb1d5672/hparams/MEMIT_ARE/Llama3-8B-Instruct.json#L22

Logs:
<img width="1792" height="1137" alt="Image" src="https://github.com/user-attachments/assets/eb420c48-aa07-4d7a-833b-740e48a6efb2" />

**Comment by jasonrichdarmawan** [comment]
oh, it's layer 4. That's why the z error is high

**Comment by jasonrichdarmawan** [comment]
~~z error is high because `target = target_init + delta`. however, now what worries is me that the `z error` is not 0 when i remove the `delta`~~

<img width="1792" height="1137" alt="Image" src="https://github.com/user-attachments/assets/933673a5-3278-4eb6-bc72-6756e5b3d671" />

**Comment by jasonrichdarmawan** [comment]
~~`z_error` is not 0 when `delta` is removed, maybe because of RoPE?~~

just learnt about RoPE. shouldn't affect it, right?

~~is it because causal_mask is not implemented because attention_mask is not passed?~~

llama creates the causal mask. so i am confused why it's not identical
